helix-core/src/main/java/org/apache/helix/manager/zk/ZkCacheBaseDataAccessor.java

```diff
@@ -842,4 +894,116 @@ public void close() {
       _zkClient.close();
     }
   }
+
+  public static class Builder {
```

Comment:
1) This Builder code is very similar across all Helix APIs, I do recommend us to find some way to consolidate them. 
2) For existing use case (who do not care about realm, or who is not  going to shard to multiple ZK), do they need to change anything in their code?

Reply:
@lei-xia 
1. See https://github.com/apache/helix/issues/873 . We will do this once all Helix Java APIs have been implemented since this is a purely *internal* implementation detail. User-facing APIs stay the same.

2. No, please see existing constructors for the fallback logic.